src/Microsoft.DotNet.XHarness.CLI/Commands/Android/AndroidAdbCommand.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.DotNet.XHarness.Android;
+using Microsoft.DotNet.XHarness.CLI.CommandArguments.Android;
+using Microsoft.DotNet.XHarness.Common;
+using Microsoft.DotNet.XHarness.Common.CLI;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.XHarness.CLI.Commands.Android;
+
+internal class AndroidAdbCommand : XHarnessCommand<AndroidAdbCommandArguments>
+{
+    private const string Description = "Invoke bundled adb with given arguments";
+    protected override string CommandUsage { get; } = "android adb [OPTIONS] -- [ADB ARGUMENTS]";
+    protected override string CommandDescription => Description;
+
+    protected override AndroidAdbCommandArguments Arguments { get; } = new();
+
+    public AndroidAdbCommand() : base(TargetPlatform.Android, "adb", false, new ServiceCollection(), Description)
+    {
+    }
+
+    protected override Task<ExitCode> InvokeInternal(ILogger logger)
+    {
+        if (!PassThroughArguments.Any())
+        {
+            logger.LogError("Please provide delimeter '--' followed by arguments for ADB:" + Environment.NewLine +
+                $"    {CommandUsage}" + Environment.NewLine +
+                $"Example:" + Environment.NewLine +
+                $"    android adb --timeout 00:01:30 -- devices -l");
+
+            return Task.FromResult(ExitCode.INVALID_ARGUMENTS);
+        }
+
+        var runner = new AdbRunner(logger);
+
+        try
+        {
+            var result = runner.RunAdbCommand(string.Join(" ", PassThroughArguments), Arguments.Timeout);
```

Comment:
Does it handle white space in the arguments correctly? For example if I call `xharness adb -- "some arg with spaces"` it should pass it as a single argument.

Reply:
Good point. I blindly trusted the existing Android code but now I verified it and we were not doing this so I added quotation from Apple side (we unfortunately have different process managers 😥 for some reason)

Reply:
I think the most correct way to handle this would be to keep the arguments as an array and use `ProcessStartInfo.ArgumentList` instead of `Arguments`. It's more errorproof than making sure that everything is quoted correctly.

Reply:
Oooh, I was too fast and didn't even think for a second `arguments` is a `string` 😅

I see the failures in tests now:
```
Executing command: 'adb.exe  "devices -l"'
```

Reply:
I think the most correct way would be to switch to a single implementation in this repo which would be the one coming from Xamarin

Reply:
@lpatalas I switched `AdbRunner` to accept an array of arguments instead. It uncovered many places where the code wasn't escaping potential problems in paths and other places.